src/webportal/src/app/job-submission/components/tools/ssh-generator.jsx

```diff
@@ -0,0 +1,218 @@
+        false,
+        0,
+        null,
+      );
+    }
+    aLink.dispatchEvent(event);
+  };
+
+  return (
+    <Dialog
+      hidden={!isOpen}
+      onDismiss={handleCancel}
+      dialogContentProps={{
+        type: DialogType.normal,
+        styles: {
+          title: { padding: '20px 36px 12px 20px' },
+          inner: { padding: '0px 40px 20px 20px' },
+          topButton: { padding: '20px 20px 0px 0px' },
+        },
+        title: (
+          <span
+            className={c(t.mb2, t.fw6, FontClassNames.semibold)}
+            style={{ fontSize: FontSizes.icon }}
+          >
+            SSH Key Generator
+          </span>
+        ),
+      }}
+      minWidth={900}
+      maxWidth={900}
+      modalProps={{
+        isBlocking: false,
+        styles: { main: { maxWidth: 900 } },
+      }}
+    >
+      <Hint>
+        Please download SSH private key then click <b>Use</b> button to use this
+        key pair in job. Then after job submitted, you can ssh to job containers
+        as user root with the downloaded private key through container ip and
+        ssh port. Try using{' '}
+        <code>
+          {'ssh -i <private key path> -p <ssh port> root@<container ip>'}
+        </code>{' '}
+        to ssh to job container.
+      </Hint>
+
+      <Separator>SSH key pair</Separator>
+
+      <Stack
+        horizontal
+        tokens={{ childrenGap: 50 }}
+        styles={{ root: { width: 860 } }}
+      >
+        <Stack tokens={{ childrenGap: 10 }} grow={1}>
+          <TextField
+            label='Private Key'
+            multiline
+            rows={20}
+            readonly
+            defaultValue={isNil(sshKeys) ? '' : sshKeys.private}
+          />
+          <DefaultButton
+            onClick={ev => {
+              setDownloadedPriKey(true);
+              downloadAsFile(sshKeys.private, 'id_rsa_pai', ev);
+            }}
+          >
+            Download Private Key
+          </DefaultButton>
+        </Stack>
+        <Stack tokens={{ childrenGap: 10 }} grow={1}>
+          <TextField
+            label='Public Key'
+            multiline
+            rows={20}
+            readonly
+            defaultValue={isNil(sshKeys) ? '' : sshKeys.public}
+          />
+          <DefaultButton
+            onClick={ev => downloadAsFile(sshKeys.public, 'id_rsa_pai.pub', ev)}
+          >
+            Download Public Key
+          </DefaultButton>
+        </Stack>
+      </Stack>
+
+      <Separator></Separator>
+
+      <Stack
+        horizontal
+        horizontalAlign='space-evenly'
+        tokens={{ childrenGap: 150 }}
+        styles={{ root: { width: 860 } }}
+      >
+        <Stack.Item align='end'>
+          {!downloadedPriKey && (
+            <Label required={true}>
+              Download private key before use ssh keys in job!
+            </Label>
+          )}
+          <DefaultButton
+            align='baseline'
+            onClick={handleConfirm}
+            disabled={!downloadedPriKey}
```

Comment:
If user clicks download and then cancel, the use button will be still enabled

Reply:
I think this is user action and user should responsible for this. Or exp team could refine the UE later